src/main.rs

```diff
@@ -87,168 +82,120 @@ async fn floodfill_pt(data: web::Data<AppState>, input: web::Json<UserInputJSON>
-            .collect();
-        println!("Parallel floodfill took {:?}", now.elapsed());
-    } else {
-        let count_original_nodes: u32 = graph_walk_guard.len() as u32;
+    serde_json::to_string(&results).unwrap()
+}
+
+// No changes to the graph allowed
+fn floodfill_pt_no_changes(data: web::Data<AppState>, input: web::Json<UserInputJSON>) -> String {
+    println!("Floodfill request received, without changes");
+    let time_of_day_ix = get_time_of_day_index(input.trip_start_seconds);
+
+    let (node_values_1d, graph_walk, graph_pt, node_values_padding_row_count) =
+        read_files_serial_excluding_travel_time_relationships_and_subpurpose_lookup(input.year);
 
-        let graph_walk_unguarded = &*graph_walk_guard;
-        let graph_pt_unguarded = &*graph_pt_guard;
-        let node_values_1d_unguarded = &*node_values_1d_guard;
+    println!("Got files read in for {}", input.year);
 
-        println!("Creating tuples to pass to floodfill for 2022 data");
-        for i in 0..input.start_nodes_user_input.len() {
-            model_parameters_each_start.push((
-                graph_walk_unguarded, //&data.graph_walk,
-                NodeID(input.start_nodes_user_input[i] as u32),
-                node_values_1d_unguarded, //&data.node_values_1d,
+    println!(
+        "Started running floodfill\ttime_of_day_ix: {}\tNodes count: {}",
+        time_of_day_ix,
+        input.start_nodes_user_input.len()
+    );
+    let now = Instant::now();
+    let indices = (0..input.start_nodes_user_input.len()).collect::<Vec<_>>();
+    let results: Vec<(i32, u32, [i64; 32], Vec<u32>, Vec<u16>)> = indices
+        .par_iter()
+        .map(|i| {
+            floodfill(
+                &graph_walk,
+                &graph_pt,
+                NodeID(input.start_nodes_user_input[*i] as u32),
+                &node_values_1d,
                 &data.travel_time_relationships_all[time_of_day_ix],
                 &data.subpurpose_purpose_lookup,
-                graph_pt_unguarded, //&data.graph_pt.lock().unwrap(),
                 input.trip_start_seconds,
-                Cost(input.init_travel_times_user_input[i] as u16),
-                count_original_nodes,
-                *&data.node_values_padding_row_count,
+                Cost(input.init_travel_times_user_input[*i] as u16),
+                graph_walk.len() as u32,
+                node_values_padding_row_count,
                 &input.target_destinations,
-            ))
-        }
-
-        println!(
-            "Started running floodfill\ttime_of_day_ix: {}\tNodes count: {}",
-            time_of_day_ix,
-            model_parameters_each_start.len()
-        );
-        let now = Instant::now();
-        parallel_res = model_parameters_each_start
-            .par_iter()
-            .map(|input| floodfill(*input))
-            .collect();
-        println!("Parallel floodfill took {:?}", now.elapsed());
-    }
-
-    if input.new_nodes_count > 0 {
```

Comment:
All of the logic to "reset" the modified objects goes away, since each request will just deep-clone and then mutate.

I haven't evaluated the perf impact of this.